storagenode/storagenodedb/infodb.go

```diff
@@ -43,12 +44,22 @@ func newInfo(path string) (*InfoDB, error) {
 
 // NewInfoInMemory creates a new inmemory InfoDB.
 func NewInfoInMemory() (*InfoDB, error) {
-	db, err := sql.Open("sqlite3", ":memory:")
+	// create memory DB with a shared cache and a unique name to avoid collisions
+	db, err := sql.Open("sqlite3", fmt.Sprintf("file:memdb%d?mode=memory&cache=shared", rand.Int63()))
 	if err != nil {
 		return nil, ErrInfo.Wrap(err)
 	}
 
-	dbutil.Configure(db, mon)
+	// Set max idle and max open to 1 to control concurrent access to the memory DB
+	// Setting max open higher than 1 results in table locked errors
+	db.SetMaxIdleConns(1)
+	db.SetMaxOpenConns(1)
+	db.SetConnMaxLifetime(-1)
```

Comment:
Setting the `MaxOpenConns` to 1 is basically the same thing as having a lock around every transaction.

I think the appropriate thing to do is to, instead of removing the `Mutex` entirely, change it to an `RWMutex` and acquire the appropriate lock. Then we can avoid the 10s busy timeout when the database is backed by a file on disk, as well.

Reply:
These are for the in memory DB only...Which to my knowledge is only used by tests.  But maybe I missed something

Reply:
Indeed, it is only used in tests, but it'd be nice the tests matched the semantics of what production is as close as possible. We're not actually testing that concurrent access works fine, anymore.

Reply:
I take it back. An `RWMutex` is exclusive to readers while a writer is active, which is not what we want. It's also insufficient to just lock on writes because memory mode does not have a WAL so writes and reads must be exclusive. So either we have different locks on different code paths, accept this as is, or attempt to use a file for the tests. I'd prefer not to have different code paths.